### ISSUE TITLE
Make the odometer suitable as a long-term statistic

### ISSUE THREAD

**Issue by mthmulders** [skodaconnect/homeassistant-skodaconnect#213]
**Is your feature request related to a problem? Please describe.**
I'd love to see a chart of the mileage of my Skoda car. I thought I would need a [Statistics Graph Card](https://www.home-assistant.io/dashboards/statistics-graph/) for that, but it doesn't show the odometer as a source. 

**Describe the solution you'd like**
The [Expected data source not listed](https://www.home-assistant.io/more-info/statistics/) page suggests that the `state_class` of the entity may not be correct.

**Describe alternatives you've considered**
n/a.

**Additional context**
I'm quite a starter when it comes to using Home Assistant - so if my understanding of how HA or this Skoda integration works is insufficient, I'm definitely open for alternative approaches.

**Comment by dvx76** [comment]
Seems reasonable. We're currently setting `state_class` for `'battery_level', 'adblue_level', 'fuel_level', 'charging_time_left', 'charging_power', 'charge_rate', 'electric_range', 'combustion_range', 'combined_range', 'outside_temperature'`. (see https://github.com/skodaconnect/homeassistant-skodaconnect/blob/6bde424e086ffde3b051e83c6f1f80a4c425dc44/custom_components/skodaconnect/sensor.py#L65)

Settings `state_class = total` for the appropriate sensors (like odometer) seems valid. Setting state class does result in long-term statistics being stored for it.

Note that if all you want is a graph with recent values I'd recommend looking into [mini graph card](https://github.com/kalkih/mini-graph-card) and [apex charts card](https://github.com/RomRider/apexcharts-card), which aren't tied into the native HA statistics stuff.

E.g. minigraph with the odometer sensor (so without `state_class`):

![image](https://github.com/skodaconnect/homeassistant-skodaconnect/assets/785424/37c37f71-a010-46ff-96d6-47ebcb799cbb)

```
type: custom:mini-graph-card
hours_to_show: 168
entities:
  - sensor.tmxxxxxxxx_odometer
```

**Comment by CaptSB** [comment]
I would like to be able to plot a statistics bar chart of the daily mileage based on the odometer reading but I can't plot the "change" in a statistics chart, it only gives the total Odometer. I suspect this is what the original poster was also referring to. If there is a way of doing this with the current code in HA then I am happy to learn how to do it.

**Comment by dvx76** [comment]
@CaptSB , as mentioned previously, take a look at the ApexCharts card.

```yaml
type: custom:apexcharts-card
graph_span: 7d
series:
  - entity: sensor.tmxxxxxxxxx_odometer
    type: column
    group_by:
      func: diff
      duration: 1day
```

![image](https://github.com/skodaconnect/homeassistant-skodaconnect/assets/785424/e28e7682-b940-4880-a6cc-05bd84e908f4)

@MaZa78 @mthmulders , setting `state_class: total_increasing` for the odo sensor sounds like the right thing to do. I'll take a look when I have time but feel free to submit a PR.

**Comment by CaptSB** [comment]
That is perfect!

I now have the daily mileage chart I wanted!

Many thanks

Richard

**Comment by haraldpaulsen** [comment]
Is this request implemented now? I read from the pull request that it is merged, but my odometer does not have long term statistics.

Running version 1.2.12

It was merged into main on 13th of april, and it was listed as part of 1.2.10, but my inspection says:

```
model: Škoda CitigoE iV STYLE/2019-12-05
unit_of_measurement: km
assumed_state: true
device_class: distance
icon: mdi:speedometer
friendly_name: Oda Odometer
```

Where is the state_class?

**Comment by dvx76** [comment]
Yeah I think the fix is wrong. The attr for the odometer sensor seems to be `distance`, not `odometer` https://github.com/skodaconnect/skodaconnect/blob/75088af7795aa049dafb8442bf9dc0b5d817cdc7/skodaconnect/dashboard.py#L1196

**Comment by WebSpider** [comment]
This integration is no longer under active development. Please consider switching to https://github.com/skodaconnect/homeassistant-myskoda
